Fix Google Calendar caching when offline

PhilRW opened the pull request:
## Description:

Events from Google Calendar were not firing under the following circumstances:

1. ha starts as normal with Google Calendar configured as per instructions
1. ha loses network connectivity to Google
1. ha attempts update of Google Calendar
1. calendar/google component throws uncaught Exception causing update method to not return
1. cached Google Calendar event does not fire, remains "Off"

Catching the Exception and returning `False` from the `update()` method causes the correct behavior (i.e., the calendar component firing the event as scheduled using cached data).

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

Commit:
Fix Google Calendar caching when offline

Events from Google Calendar were not firing under the following
circumstances:

1. Start ha as normal with Google Calendar configured as per
   instructions.
2. ha loses network connectivity to Google
3. ha attempts update of Google Calendar
4. calendar/google component throws uncaught Exception causing update
   method to not return
5. (cached) Google Calendar event does not fire, remains "Off"

Catching the Exception and returning False from the update() method
causes the correct behavior (i.e., the calendar component firing the
event as scheduled using cached data).

Commit:
Add requirements

balloob commented on homeassistant/components/calendar/google.py:
```diff
@@ -14,6 +14,8 @@
     GoogleCalendarService)
 from homeassistant.util import Throttle, dt
 
+REQUIREMENTS = ["httplib2==0.10.3"]
```
Don't add this. Requirements are installed as part of the parent component.

PhilRW replied:
I don't understand where I'm supposed to add it. In the `__init__.py` for the `calendar` component?

PhilRW replied:
If I don't add it, the tox unit testing will fail since the `gen_requirements_all.py` script seems to only add test requirements from the `TEST_REQUIREMENTS` tuple if they're actually listed somewhere in a module.

PhilRW replied:
Unless you'd rather I edit `requirements_test.txt` instead?

balloob replied:
The requirement that depends on this package needs to be whitelisted.

balloob replied:
Alternatively, you can use the `MockDependency` decorator from `tests.common`.

balloob commented on homeassistant/components/calendar/google.py:
```diff
@@ -62,7 +64,14 @@ def __init__(self, calendar_service, calendar_id, search=None):
+            service = self.calendar_service.get()
+        except ServerNotFoundError:
+            _LOGGER.warning("Unable to connect to Google, using cached data")
+            return False
```
The return value is not used, so remove it.

PhilRW replied:
What about [this code](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/calendar/__init__.py#L138)?

PhilRW replied:
...or do you mean that the `not self.data.update()` check is looking for `None` instead of `False`?

balloob replied:
oh, didn't realize calendar is using their own stuff. kinda makes sense. Ok in that case it's fine.

balloob commented on tests/components/calendar/test_google.py:
```diff
@@ -369,8 +359,7 @@ def test_all_day_offset_event(self, mock_next_event):
             'end': {
-                'date': (tomorrow + dt_util.dt.timedelta(days=1))
-                .isoformat()
+                'date': (tomorrow + dt_util.dt.timedelta(days=1)).isoformat()
```
Try not to clean up the code unless you're actually touching it, it makes the diff of actual changes harder to read.

Commit:
Revert code cleanup

Commit:
Remove explicit return value from update()

Commit:
Revert "Remove explicit return value from update()"

This reverts commit 7cd77708af658ccea855de47a32ce4ac5262ac30.

Commit:
Use MockDependency decorator

No need to whitelist google-python-api-client for a single unit test at
this point.
